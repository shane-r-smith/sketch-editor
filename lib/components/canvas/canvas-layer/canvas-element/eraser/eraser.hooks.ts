import type { LineConfig } from "konva/lib/shapes/Line";
import { useMemo } from "react";
import type { CanvasElementProps } from "../canvas-element.types";

export const createEraser = ({
  element,
  ...rest
}: CanvasElementProps): LineConfig => ({
  ...rest,
  ...element,
  tension: 0.5,
  lineCap: "round",
  lineStroke: "round",
  globalCompositeOperation: "destination-out",
});

export const useEraser = (props: CanvasElementProps): LineConfig => {
  return useMemo<LineConfig>(() => {
    return createEraser(props);
  }, [props]);
};
