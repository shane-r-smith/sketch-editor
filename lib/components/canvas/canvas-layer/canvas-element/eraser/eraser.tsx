import { Line } from "react-konva";
import { useEraser } from "./eraser.hooks";
import type { CanvasElementProps } from "../canvas-element.types";

export function Eraser(props: CanvasElementProps) {
  const eraser = useEraser(props);

  return <Line {...eraser} />;
}
