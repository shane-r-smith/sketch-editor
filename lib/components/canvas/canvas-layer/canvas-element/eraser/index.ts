export * from "./eraser";
export * from "./eraser.hooks";
