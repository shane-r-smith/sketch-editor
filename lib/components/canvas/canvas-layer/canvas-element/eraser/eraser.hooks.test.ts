import { describe, it, expect } from "vitest";
import { createEraser, useEraser } from "./eraser.hooks";
import type { CanvasElementProps } from "../canvas-element.types";
import { renderHook } from "@testing-library/react";

// Mock props for testing
const baseProps: CanvasElementProps = {
  element: {
    key: "ERASER",
    points: [0, 0, 10, 10],
    stroke: "#000000",
    strokeWidth: 10,
  },
  x: 5,
  y: 5,
  id: "eraser",
};

describe("createEraser", () => {
  it("returns correct LineConfig for eraser", () => {
    const result = createEraser(baseProps);
    expect(result).toMatchSnapshot();
  });
});

describe("useEraser", () => {
  it("returns memoized LineConfig", () => {
    const { result } = renderHook(
      (props: CanvasElementProps) => useEraser(props),
      { initialProps: baseProps }
    );
    expect(result.current).toMatchSnapshot();
  });
});
