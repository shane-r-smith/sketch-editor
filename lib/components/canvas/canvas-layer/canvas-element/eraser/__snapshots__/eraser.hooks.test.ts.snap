// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createEraser > returns correct LineConfig for eraser 1`] = `
{
  "globalCompositeOperation": "destination-out",
  "id": "eraser",
  "lineCap": "round",
  "lineStroke": "round",
  "points": [
    0,
    0,
    10,
    10,
  ],
  "stroke": "#000000",
  "strokeWidth": 10,
  "tension": 0.5,
  "tool": "ERASER",
  "x": 5,
  "y": 5,
}
`;

exports[`useEraser > returns memoized LineConfig 1`] = `
{
  "globalCompositeOperation": "destination-out",
  "id": "eraser",
  "lineCap": "round",
  "lineStroke": "round",
  "points": [
    0,
    0,
    10,
    10,
  ],
  "stroke": "#000000",
  "strokeWidth": 10,
  "tension": 0.5,
  "tool": "ERASER",
  "x": 5,
  "y": 5,
}
`;
